signer/universal/universal.go

```diff
@@ -110,21 +120,95 @@ func NewSigner(root Root, policy *config.Signing) (signer.Signer, error) {
 			// of valid signers when PKCS #11 is turned
 			// on.
 			for _, possibleSigner := range localSignerList {
-				s, shouldProvide, err = possibleSigner(&root, policy)
+				s.Local, shouldProvide, err = possibleSigner(&root, policy)
 				if shouldProvide {
 					break
 				}
 			}
 
-			if s == nil {
+			if s.Local == nil {
 				err = cferr.New(cferr.PrivateKeyError, cferr.Unknown)
 			}
 		}
 
 		if policy.NeedsRemoteSigner() {
-			s, err = remote.NewSigner(policy)
+			s.Remote, err = remote.NewSigner(policy)
+		}
+	}
+
+	return &s, err
+}
+
+// getMatchingProfile returns the SigningProfile that matches the label passed.
+func (s *Signer) getMatchingProfile(label string) (*config.SigningProfile, error) {
```

Comment:
Maybe `label` should be renamed `profile` --- the terminology is admittedly confusing, but profile is the more accurate term here.